### DISCUSSION TITLE
Code Annotation number in text [PDF]

### DISCUSSION THREAD

**Discussion by sasja-san** [quarto-dev/quarto-cli#11461] [Q&A]
### Description

I currently have the kind of writing style seen in the picture. But I've been told I need to change uncommented blocks of code into numbered listings with captions.
I didn't succeed in creating line breaks within captions so I was thinking that perhaps there's a way to write the circled numbers within my caption text? Because the circled numbers will be generated from the comments in the code, regardless if the numbered list exists below it or not. I just want to be able to use those circled numbers in a way that fits my requirements.

![code-annotation-nums](https://github.com/user-attachments/assets/bf96cb18-ff42-4b67-b765-1a3bde1809b4)

**Comment by cderv** [comment]
Do I understand correctly that you don't want to use the code annotation feature the way quarto does it but you want to leverage the fact that we keep the circled number and then use them yourself in a caption ? 

Did you look into the .tex code quarto generates to see if it would work to reuse it in a caption ? 

It seems to me this is a latex how question, which would require some expertise outside of quarto. If you manage to come with a nice formatting, maybe we could add this as new option to set annotation as caption 😅

**Reply by mcanouil** [reply]
FYI, Quarto uses the following LaTeX command for the number:

```latex
\newcommand*\circled[1]{
  \tikz[baseline=(char.base)]{
    \node[shape=circle,draw,inner sep=1pt] (char) {{\scriptsize#1}};
  }
}
```

Indeed, this seems to be more of a LaTeX question than a Quarto question.

**Reply by sasja-san** [reply]
It looked promising, but sadly it didn't work.

I'll work around it by only using the \circled command inside of my main text instead. Thanks!

![quarto-circled-num-error](https://github.com/user-attachments/assets/9512079b-6408-43e1-af4a-5e0c33da880d)

**Reply by sasja-san** [reply]
Sorry, the rendering with the caption location oddity was due to a comment within the `lst` block. This is the actual error.

![quarto-circled-num-error](https://github.com/user-attachments/assets/b6c4a4b6-bb11-4b5d-8667-792d687af251)

Once I finish the work with https://github.com/quarto-dev/quarto-cli/issues/11384 I might look at this as well.

**Reply by mcanouil** [reply]
It might require additional LaTeX packages but it gives you the general idea of the trick.
It's really all about LaTeX now.

**Reply by cderv** [reply]
You need at least tikz package for sure so that tikz command is found. 

Quarto adds it as a LaTeX dependency when code annotation is used, but if you opt out you need to add it yourself in your document.
